chain/chain/src/stateless_validation/chunk_endorsement.rs

```diff
@@ -60,8 +58,12 @@ pub fn validate_chunk_endorsements_in_block(
             }
             continue;
         }
+
         // Validation for chunks in each shard
         // The signatures from chunk validators for each shard must match the ordered_chunk_validators
+        let shard_id = chunk_header.shard_id();
+        let shard_index = shard_layout.get_shard_index(shard_id);
```

Comment:
I know it's probably my code and it likely is correct because we only have new chunks here but anyway this pattern makes me uneasy. Even if only because someone may copy paste is somewhere where the assumption is not valid. I'll have another go at those and fix it, I suppose there is more than one such case here.